bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -2256,15 +2258,20 @@ public void setBackground(Color color) {
 	device.internal_dispose_GC(hDC, null);
 }
 
-private void setCurrentDeviceZoom(int newZoomFactor) {
-	if (this.currentDeviceZoom != newZoomFactor) {
-		this.currentDeviceZoom = newZoomFactor;
+private void setCurrentNativeZoom(int newNativeZoom) {
+	int oldZoom = this.getZoom();
+	if (oldZoom != newNativeZoom) {
```

Comment:
I still don't see my comment (https://github.com/eclipse-platform/eclipse.platform.swt/pull/1229#discussion_r1611590128) properly addressed: the current implementation does still not always update the `currentNativeZoom` with the `newNativeZoom` when this method is called, which I find confusing. This setter is only called when the current and old current zoom values differ anyway, so couldn't the `currentNativeZoom` simply always be updated when this method is called?
In addition, the modification of the condition (now comparing against the new native zoom instead of the new current zoom) changes when the `height` and `width` value will be reset. Is this functional change intended? If yes, the comment needs to be updated.